scripts/js-analyze.js

```diff
@@ -727,7 +727,13 @@ let Analyzer = {
 
     case "ObjectPattern":
       for (let prop of pat.properties) {
-        this.pattern(prop.value);
+        if ("value" in prop) {
```

Comment:
Could this instead be a check if (prop.type == "Property")? Looking at ASTSerializer::objectPattern() (I couldn't find any better documentation of this...), that seems to be the only other case. That feels a little easier to understand than checking for some property.

Reply:
Yup, makes sense. I updated the PR to do that. Also added a test.